### ISSUE TITLE
Question: where I can find some documentation?

### ISSUE THREAD

**Issue by raffaeler** [abbaye/WpfHexEditorControl#75]
I looked at the samples but I spent one hour just to understand that `Stream` was the property that I needed to set to show something.
The control is awesome, but without some documentation I bet most of the people is not using at its best.

Thanks for the great project.

**Comment by abbaye** [comment]
Thank you 😊 I made this control in my free time.

You can check samples for help. But it's a good idea to make a documentation with pictures. I will work on that's this fall

**Comment by raffaeler** [comment]
I totally understand. I don't mean a full featured documentation, but something that can make us understand the main use cases and what do the properties mean.

I also suggest to create more small samples rather than one big. I believe the use-cases should drive them and be easier for you as well.

Thanks!

**Comment by abbaye** [comment]
It's a nice idea. I will going on this way (more little sample). 

(Sorry for my English)

**Comment by abbaye** [comment]
If you want put a 🌟 on my project

**Comment by reproteq** [comment]
Hi .
Thanks for this contribution, I really like it.
But I think it takes some information to be able to make some changes, methods, properties, etc ...
I also need to access buffer byte []
var x = this.FileTab.Items.Count.ToString (); // tab counter
var xx = this.FileTab.SelectedIndex.ToString (); // index selected
 var xxx = this.FileTab.SelectedValue.ToString (); // tab filename
 string ToTblString (byte [] data)
class ByteProvider: IDisposable
class PathToFilenameConverter: IValueConverter 
Path.GetFileName (filename);


I don't know some information would be fine because it looks like a good tool but for example I would like it to be able to eat 2 hex and save diff, to apply patch etc ...
something like this that I have done recently.
https://github.com/reproteq/DiffPatchWpf

a binary comparer that saves diff to a txt and can apply these changes to another binary.
You could also implement it like this almost better, if you want I'll help you.
thank you very much for your time and work

**Comment by abbaye** [comment]
Hi @reproteq 

Thanks for you good words 😁

What you want/needs exactly ?

- You want the hexeditor can compare to files and save the différence on another files ?

-you need somes information about some properties and methods for use the hexeditor?

Just explain in detail what you need please 🙂

Best regards

**Comment by reproteq** [comment]
I need to compare 2 files and save by lines a patch.txt file with addresses and values.
then I can open that file and apply the differences it contains.
something similar to what I have done in another project called wpfdiffpatch.
Could you tell me how to get the array byte [] data buffer from the file loaded in the selected tab?
FileTab.SelectedIndex
and the path off file load

**Comment by abbaye** [comment]
You can use for this:

Hexeditor.GetAllBytes();

if you want that your différence appears, you can use 2 hexeditor and use the CustomBackgroundBlock class in the editor to put byte in incidence. If you need a sample for do that I can make one for you

**Comment by reproteq** [comment]
You could also make the byteperline length of the line draggable, which is adjusted by stretching or collapsing the window.
You could make it compare 2 files and align it in the main view the 2 at the same time and highlight the changes and that the lines are synchronized to be able to lower or raise them at the same time
https://www.youtube.com/watch?v=EpyuF4t5MWk

**Comment by abbaye** [comment]
All its possible but I need the time for do that :) 

I make the hexeditor in my free time.

But I can make a little sample for help you if you need

**Comment by reproteq** [comment]
ok , thanks  if you can help me .
this is mi fork  and custom
https://github.com/reproteq/WpfHexEditorControl/blob/master/Sources/Samples/HexEditReproteq/MainWindow.xaml.cs
y add menu with 2 buttons for make comparation and save diff.txt and other button for load this diff.txt and apply in other file loaded in app.

and this is mi litle app  for make patchs.
https://github.com/reproteq/DiffPatchWpf/blob/main/MainWindow.xaml.cs

**Comment by abbaye** [comment]
Good I will check this.

I will make a new sample directly  in my repo. 

You will can take the code when is done

**Comment by reproteq** [comment]
ok thank in advance .
i go to sleep here 1 am.
by bro have nice n8

**Comment by abbaye** [comment]
No trouble here is only 7pm

**Comment by abbaye** [comment]
@reproteq take the lastest code on github and launch the sample WpfHexEditor.Sample.BinaryFilesDifference.

It's not perfert and completed but i hope it will help you.

regards

**Comment by reproteq** [comment]
ohh yeahh thanks alot mate i go test this changes.
have nice day bro.

**Comment by abbaye** [comment]
It's not perfect now I'm coded this in about 2 hours but I think is a nice sample.

I will optimize and complete this this week if possible for me

**Comment by reproteq** [comment]
thanks for your time . I'm looking at it but I see some variables and properties in red I will have to take a good look at it to make it go. it would be a good contribution to your application to add these functions to the master project.
Thanks again for your time I will be aware of your improvements and have a good day friend

**Comment by abbaye** [comment]
Make me a pull request for function you want to contribute. :)

**Comment by abbaye** [comment]
Humm clean your project and rebuild.

I don't have variable with error in my visual studio.

**Comment by abbaye** [comment]
@reproteq get the new code and test the sample please 😄

**Comment by reproteq** [comment]
hi mate good job very beautiful.
now next step is save diff in txt file , 1 line x change.
after load file txt and apply this diff in other file loaded.


![image](https://user-images.githubusercontent.com/16064306/103573810-29b72f80-4ecf-11eb-936b-6f8ea03fe693.png)

but  mi solution show  25 errors.

great job mate.

**Comment by abbaye** [comment]
For error make a clean in solution after that close visual studio, reopen solution and rebuild.

**Comment by abbaye** [comment]
I will implement save and patch after this

**Comment by reproteq** [comment]
yes mate work after clean solution, close vs and build.
thanks alot  is very very beatiful.
great job mate

**Comment by abbaye** [comment]
Only ~2-3hr of job for this result

Thank you mister

**Comment by reproteq** [comment]
Your are a great developer man, and you know alls class , methods and others ... for make a fast job in this project.
Incredible fast results  and very very beatyfull interface.
Diff color i think ever read not rand , jj , thanks other time mate.
Can you continue same , step to step ,this is a good sample for alls.

**Comment by reproteq** [comment]
I'm reading your code and trying to understand well how it is built to be able to make any type of adjustment or modification and not have to waste more time, I really appreciate it brother.
It would have been hard for me to get to where you are at this moment and the truth is that before contacting you I had already spent many hours trying to understand the system in order to work well with it.
I really like how it's written so I chose this base project to add this kind of functionality.
I use 3 different softwares to do what this one is doing, wonderful
